src/liballoc/heap.rs

```diff
@@ -16,6 +16,18 @@
 
 use core::{isize, usize};
 
+#[allow(improper_ctypes)]
+extern {
+    #[allocator]
```

Comment:
Huh, well I'm maybe more confused now. Does `#[allocator]` mean two different things depending on context? It seems like it does, and they are both gated by the `allocator` feature? Might we rename one of these so they aren't overloaded. I guess it doesn't matter right now as long as they are gated....